src/config/env/index.ts

```diff
@@ -0,0 +1,19 @@
+import "dotenv/config";
+
+import { z } from "zod";
+
+const envSchema = z.object({
+  DISCORD_TOKEN: z.string(),
+  CLIENT_ID: z.string(),
+  DISCORD_SERVER_ID: z.string(),
+});
+
+const _env = envSchema.safeParse(process.env);
+
+if (!_env.success) {
+  console.error("invalid environment variables", _env.error.format());
+
+  throw new Error("Invalid environment variables");
+}
+
+export const env = _env.data;
```

Comment:
não sei se vale a pena usar o zod só pra tipar as envs, [eu fiz isso nesse arquivo usando uma interface](https://github.com/devhatt/hatbot-discord/blob/master/process-env.d.ts), mas é discutivel, se com zod for melhor a gente pode deixar

Reply:
eu gosto porque a gente consegue validar se o usuário realmente está com as variáveis, enfrentei isso durante a execução do projeto, caso não tenha as variáveis já é lançado o para isso, sem essa validação o usuário recebe o erro do discord.js ("token invalid")

Reply:
isso é interessante, vamos usar o zod entao e exclui o arquivo .d.ts que eu criei pras envs.

se eu nao me engano em algum comando eu precisei usar dotenv tambem. se puder dar uma olhada e mudar.

Reply:
feito